relentless/potential/potential.py

```diff
@@ -34,28 +131,123 @@ def evaluate(self, pair):
+            raise KeyError('The type pairs in the data are not exactly the pairs in self')
+        for key in data:
+            key_tup = eval(key)
+            if sorted(self.params) != sorted(data[key].keys()):
+                raise KeyError('The parameters in the data are not exactly the parameters in self')
+            for param in data[key]:
+                var_data = data[key][param]
+                var_self = self[key_tup][param]
+                if var_self is None:
+                    if var_data['type'] == 'variable':
+                        self[key_tup][param] = Variable(value=var_data['value'], const=var_data['const'],
+                                                        low=var_data['low'], high=var_data['high'])
+                    elif var_data['type'] == 'scalar':
+                        self[key_tup][param] = var_data['value']
+                    else:
+                        raise TypeError('The parameter value types in the data and self do not match')
+                elif var_data['type'] == 'scalar' and isinstance(var_self, (float,int)):
+                    self[key_tup][param] = var_data['value']
+                elif var_data['type'] == 'variable' and isinstance(var_self, Variable):
+                    self[key_tup][param] = Variable(value=var_data['value'], const=var_data['const'],
+                                                    low=var_data['low'], high=var_data['high'])
+                else:
+                    raise TypeError('The parameter value types in the data and self do not match')
+
+    @classmethod
+    def from_file(cls, filename):
+        """Load data from a file into a new coefficient matrix object.
+
+        Parameters
+        ----------
+        filename : `str`
+            The name of the file from which to load data.
+
+        Returns
+        -------
+        :py:class:`CoefficientMatrix`
+            Object initialized with the data in the file.
+
+        """
+        with open(filename, 'r') as f:
+            data = json.load(f)
+        t_all = data.keys()
+        p = data[list(t_all)[0]].keys()
+        for par in t_all:
+            par = eval(par)
+            if len(par) == len(set(par)):
+                t = par
+                break
+        m = CoefficientMatrix(types=t, params=p)
+        m.load(filename)
+        return m
 
     def __setitem__(self, key, value):
-        """ Set coefficients for the (i,j) pair."""
+        """Set coefficients for the (i,j) pair."""
         for p in value:
             if p not in self.params:
-                raise KeyError('Only the known parameters can be set in coefficient matrix.')
-            self[key][p] = value[p]
+                raise KeyError('Only the known parameters can be set in the coefficient matrix.')
+        for p in self.params:
+            if p in value:
+                self[key][p] = value[p]
+            else:
+                self[key][p] = self.default[p] if p in self.default else None
```

Comment:
I think it might be better to simplify this as two update calls:
```py
self[key].update(self.default)
self[key].update(value)
```
The first resets everything to its default, then the second overwrites whatever was set in the values.